app/assets/javascripts/discourse/app/lib/desktop-notifications.js

```diff
@@ -144,44 +144,41 @@ function canUserReceiveNotifications(user) {
 
 // Call-in point from message bus
 async function onNotification(data, siteSettings, user, appEvents) {
-  if (!canUserReceiveNotifications(user)) {
-    return false;
-  }
-
-  if (!liveEnabled) {
-    return false;
-  }
+  const showNotifications = canUserReceiveNotifications(user) && liveEnabled;
+
+  if (showNotifications) {
+    const notificationTitle =
+      data.translated_title ||
+      I18n.t(i18nKey(data.notification_type), {
+        site_title: siteSettings.title,
+        topic: data.topic_title,
+        username: formatUsername(data.username),
+        group_name: data.group_name,
+      });
 
-  const notificationTitle =
-    data.translated_title ||
-    I18n.t(i18nKey(data.notification_type), {
-      site_title: siteSettings.title,
-      topic: data.topic_title,
-      username: formatUsername(data.username),
-      group_name: data.group_name,
+    const notificationBody = data.excerpt;
+    const notificationIcon =
+      siteSettings.site_logo_small_url || siteSettings.site_logo_url;
+    const notificationTag =
+      "discourse-notification-" +
+      siteSettings.title +
+      "-" +
+      (data.topic_id || 0);
+
+    await requestPermission();
+
+    // create new notification using Notifications API
+    const notification = new Notification(notificationTitle, {
+      body: notificationBody,
+      icon: notificationIcon,
+      tag: notificationTag,
     });
-
-  const notificationBody = data.excerpt;
-
-  const notificationIcon =
-    siteSettings.site_logo_small_url || siteSettings.site_logo_url;
-
-  const notificationTag =
-    "discourse-notification-" + siteSettings.title + "-" + (data.topic_id || 0);
-
-  await requestPermission();
-
-  // This shows the notification!
-  const notification = new Notification(notificationTitle, {
-    body: notificationBody,
-    icon: notificationIcon,
-    tag: notificationTag,
-  });
-  notification.onclick = () => {
-    DiscourseURL.routeTo(data.post_url);
-    appEvents.trigger("desktop-notification-opened", { url: data.post_url });
-    notification.close();
-  };
+    notification.onclick = () => {
+      DiscourseURL.routeTo(data.post_url);
+      appEvents.trigger("desktop-notification-opened", { url: data.post_url });
+      notification.close();
+    };
```

Comment:
I wonder if ideally this shouldn't be

```javascript
notification.addEventListener("click", { ... }, { once: true });
```

Reply:
Yes I think we can update, I just left the existing code in place and worked around it.. but probably worth refactoring while we are here.